src/Navigation/Notifications/Collection/Notifications.php

```diff
@@ -8,36 +8,50 @@
 namespace Friendica\Navigation\Notifications\Collection;
 
 use Friendica\BaseCollection;
-use Friendica\Navigation\Notifications\Entity;
+use Friendica\Navigation\Notifications\Entity\Notification as NotificationEntity;
 
 class Notifications extends BaseCollection
 {
-	/**
-	 * @return Entity\Notification
-	 */
-	public function current(): Entity\Notification
+	public function current(): NotificationEntity
 	{
 		return parent::current();
 	}
 
 	public function setSeen(): Notifications
 	{
-		return $this->map(function (Entity\Notification $Notification) {
-			$Notification->setSeen();
-		});
+		$class = get_class($this);
+
+		return new $class(
+			array_map(
+				function (NotificationEntity $notification) {
+					$notification->setSeen();
+				},
+				$this->getArrayCopy()
+			),
+			$this->getTotalCount(),
+		);
 	}
 
 	public function setDismissed(): Notifications
 	{
-		return $this->map(function (Entity\Notification $Notification) {
-			$Notification->setDismissed();
-		});
+		$class = get_class($this);
+
+		return new $class(
+			array_map(
+				function (NotificationEntity $notification) {
+					$notification->setDismissed();
+				},
+				$this->getArrayCopy(),
+			),
+			$this->getTotalCount(),
+		);
+
```

Comment:
As much as I like cheeky multi-level indentation, I really think this isn't necessary and the inherited `BaseCollection->map` should do the trick, but I'm still curious to hear why you went through all this trouble.

Reply:
I can create a new `mapNotifications()` method and refactor both methods to use this new method.

I would also like to get rid of the `$class = get_class($this);` and `return new $class(` code, but I have to keep it for BC. Because the `Notifications` class is not final it's possible that someone has extended this class and using `return new Notifications(` would break their code.